test/sql/function/string/test_subscript.test

```diff
@@ -0,0 +1,115 @@
+# name: test/sql/function/string/test_subscript.test
+# description: Substring test
+# group: [string]
+
+statement ok
+PRAGMA enable_verification
+
+statement ok
+CREATE TABLE strings(s VARCHAR, off INTEGER);
+
+statement ok
+INSERT INTO strings VALUES ('hello', 0), ('world', 1), ('b', 0), (NULL, 1)
```

Comment:
Could you add some tests involving unicode strings, and composite unicode characters (see e.g. `test/sql/function/string/test_complex_unicode.test`)